contracts/token/ERC1155/utils/ERC1155Utils.sol

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {IERC1155Receiver} from "../IERC1155Receiver.sol";
+import {IERC1155Errors} from "../../../interfaces/draft-IERC6093.sol";
+
+/**
+ * @dev Implementation of the ERC-1155 acceptance checks used in safe transfers.
+ * See https://eips.ethereum.org/EIPS/eip-1155
+ */
+library ERC1155Utils {
+    /**
+     * @dev Performs an acceptance check by calling {IERC1155-onERC1155Received} on the `to` address if it
+     * contains code at the moment of execution. This will revert if the recipient doesn't accept the token transfer.
+     * The call is not executed if the target address is not a contract.
+     */
+    function checkOnERC1155Received(
+        address operator,
```

Comment:
I see the point of adding `operator` to the arguments is to allow using `_msgSender()` in the ERC1155 implementation (same for ERC721), but I'm a bit concerned about making it customizable.

We should explicitly recommend `operator` to be `msg.sender` in the majority of the cases. I haven't thought of a case where you wouldn't want it like that.

Same for other `check` functions.

Reply:
Right, it's fair to say its meaning is implicit from the ERC's underlying name.

I'll make it explicit in the docs